lib/sprockets/uglifier_compressor.rb

```diff
@@ -47,14 +47,24 @@ def call(input)
       if Autoload::Uglifier::VERSION.to_i < 2
         raise "uglifier 1.x is no longer supported, please upgrade to 2.x"
       end
-      @uglifier ||= Autoload::Uglifier.new(@options.merge({ source_filename: input[:filename] }))
+      uglifier = Autoload::Uglifier.new(@options.merge(input_options(input)))
```

Comment:
> I guess it was invalidated when we started passing the filename in?

I believe it so, although the source maps of my current app seems to be working as expected even with the wrong filename being used on different assets.